### ISSUE TITLE
[Bug] Why do semantic-segmentation's labels inferred by onnx have jagged edges

### ISSUE THREAD

**Issue by AnRanbel** [open-mmlab/mmdeploy#2479]
### Checklist

- [X] I have searched related issues but cannot get the expected help.
- [X] 2. I have read the [FAQ documentation](https://github.com/open-mmlab/mmdeploy/tree/main/docs/en/faq.md) but cannot get the expected help.
- [X] 3. The bug has not been fixed in the latest version.

### Describe the bug

As shown in the figure below, the label edges inferred by the pth model are soft, but the label edges inferred by the onnx model are jagged.Thank u!
**onnxruntime**
![output_onnxruntime_0](https://github.com/open-mmlab/mmdeploy/assets/33872705/46e9591c-341d-4d0c-bdea-1899f761220c)
**pytorch**
![output_pytorch_0](https://github.com/open-mmlab/mmdeploy/assets/33872705/7f5dc59a-5047-41c7-b823-7338e683ccdf)

### Reproduction
something like this,
python tools/deploy.py \
    configs/mmseg/segmentation_onnxruntime_dynamic.py \
    unet-s5-d16_fcn_4xb4-160k_cityscapes-512x1024.py \
    fcn_unet_s5-d16_4x4_512x1024_160k_cityscapes_20211210_145204-6860854e.pth \
    demo/resources/cityscapes.png \
    --work-dir mmdeploy_models/mmseg/ort \
    --device cpu \
    --show \
    --dump-info

### Environment

```Shell
System environment:
    sys.platform: linux
    Python: 3.8.0 (default, Nov  6 2019, 21:49:08) [GCC 7.3.0]
    CUDA available: True
    numpy_random_seed: 2023
    GPU 0: NVIDIA TITAN Xp
    CUDA_HOME: /usr/local/cuda
    NVCC: Cuda compilation tools, release 11.1, V11.1.105
    GCC: gcc (Ubuntu 7.5.0-3ubuntu1~18.04) 7.5.0
    PyTorch: 1.9.0+cu111
    PyTorch compiling details: PyTorch built with:
  - GCC 7.3
  - C++ Version: 201402
  - Intel(R) Math Kernel Library Version 2020.0.0 Product Build 20191122 for Intel(R) 64 architecture applications
  - Intel(R) MKL-DNN v2.1.2 (Git Hash 98be7e8afa711dc9b66c8ff3504129cb82013cdb)
  - OpenMP 201511 (a.k.a. OpenMP 4.5)
  - NNPACK is enabled
  - CPU capability usage: AVX2
  - CUDA Runtime 11.1
  - NVCC architecture flags: -gencode;arch=compute_37,code=sm_37;-gencode;arch=compute_50,code=sm_50;-gencode;arch=compute_60,code=sm_60;-gencode;arch=compute_70,code=sm_70;-gencode;arch=compute_75,code=sm_75;-gencode;arch=compute_80,code=sm_80;-gencode;arch=compute_86,code=sm_86
  - CuDNN 8.0.5
  - Magma 2.5.2
  - Build settings: BLAS_INFO=mkl, BUILD_TYPE=Release, CUDA_VERSION=11.1, CUDNN_VERSION=8.0.5, CXX_COMPILER=/opt/rh/devtoolset-7/root/usr/bin/c++, CXX_FLAGS= -Wno-deprecated -fvisibility-inlines-hidden -DUSE_PTHREADPOOL -fopenmp -DNDEBUG -DUSE_KINETO -DUSE_FBGEMM -DUSE_QNNPACK -DUSE_PYTORCH_QNNPACK -DUSE_XNNPACK -DSYMBOLICATE_MOBILE_DEBUG_HANDLE -O2 -fPIC -Wno-narrowing -Wall -Wextra -Werror=return-type -Wno-missing-field-initializers -Wno-type-limits -Wno-array-bounds -Wno-unknown-pragmas -Wno-sign-compare -Wno-unused-parameter -Wno-unused-variable -Wno-unused-function -Wno-unused-result -Wno-unused-local-typedefs -Wno-strict-overflow -Wno-strict-aliasing -Wno-error=deprecated-declarations -Wno-stringop-overflow -Wno-psabi -Wno-error=pedantic -Wno-error=redundant-decls -Wno-error=old-style-cast -fdiagnostics-color=always -faligned-new -Wno-unused-but-set-variable -Wno-maybe-uninitialized -fno-math-errno -fno-trapping-math -Werror=format -Wno-stringop-overflow, LAPACK_INFO=mkl, PERF_WITH_AVX=1, PERF_WITH_AVX2=1, PERF_WITH_AVX512=1, TORCH_VERSION=1.9.0, USE_CUDA=ON, USE_CUDNN=ON, USE_EXCEPTION_PTR=1, USE_GFLAGS=OFF, USE_GLOG=OFF, USE_MKL=ON, USE_MKLDNN=ON, USE_MPI=OFF, USE_NCCL=ON, USE_NNPACK=ON, USE_OPENMP=ON, 

    TorchVision: 0.10.0+cu111
    OpenCV: 4.8.0
    MMEngine: 0.7.4

Runtime environment:
    cudnn_benchmark: True
    mp_cfg: {'mp_start_method': 'fork', 'opencv_num_threads': 0}
    dist_cfg: {'backend': 'nccl'}
    seed: 2023
    Distributed launcher: none
    Distributed training: False
    GPU number: 1
```

**Comment by RunningLeon** [comment]
hi, maybe your image too large? what's your env? You can run `python tools/check_env.py`.

**Comment by AnRanbel** [comment]
> hi, maybe your image too large? what's your env? You can run `python tools/check_env.py`.

thank u for your reply, when I reduced the input image size from 1181x1181 to 500x500, the above phenomenon still occurred, i.e. pytorch derived the image label with soft edges, but onnx derived the image label with jagged edges. But when I reduced again to 128x128 (the same setting as the model training), the image labels are basically the same, as shown below:
**onnx**
![output_onnxruntime_0_128](https://github.com/open-mmlab/mmdeploy/assets/33872705/955fda4b-611b-4306-9ce3-8bf423f5170c)
**pytorch**
![output_pytorch_0_128](https://github.com/open-mmlab/mmdeploy/assets/33872705/d407c86b-31d7-45d9-8340-646f8ee07a44)

*my env:*
10/08 16:42:32 - mmengine - INFO - **********Environmental information**********
10/08 16:42:33 - mmengine - INFO - sys.platform: linux
10/08 16:42:33 - mmengine - INFO - Python: 3.8.0 (default, Nov  6 2019, 21:49:08) [GCC 7.3.0]
10/08 16:42:33 - mmengine - INFO - CUDA available: True
10/08 16:42:33 - mmengine - INFO - numpy_random_seed: 2147483648
10/08 16:42:33 - mmengine - INFO - GPU 0: NVIDIA TITAN Xp
10/08 16:42:33 - mmengine - INFO - CUDA_HOME: /usr/local/cuda
10/08 16:42:33 - mmengine - INFO - NVCC: Cuda compilation tools, release 11.1, V11.1.105
10/08 16:42:33 - mmengine - INFO - GCC: gcc (Ubuntu 7.5.0-3ubuntu1~18.04) 7.5.0
10/08 16:42:33 - mmengine - INFO - PyTorch: 1.9.0+cu111
10/08 16:42:33 - mmengine - INFO - PyTorch compiling details: PyTorch built with:
  - GCC 7.3
  - C++ Version: 201402
  - Intel(R) Math Kernel Library Version 2020.0.0 Product Build 20191122 for Intel(R) 64 architecture applications
  - Intel(R) MKL-DNN v2.1.2 (Git Hash 98be7e8afa711dc9b66c8ff3504129cb82013cdb)
  - OpenMP 201511 (a.k.a. OpenMP 4.5)
  - NNPACK is enabled
  - CPU capability usage: AVX2
  - CUDA Runtime 11.1
  - NVCC architecture flags: -gencode;arch=compute_37,code=sm_37;-gencode;arch=compute_50,code=sm_50;-gencode;arch=compute_60,code=sm_60;-gencode;arch=compute_70,code=sm_70;-gencode;arch=compute_75,code=sm_75;-gencode;arch=compute_80,code=sm_80;-gencode;arch=compute_86,code=sm_86
  - CuDNN 8.0.5
  - Magma 2.5.2
  - Build settings: BLAS_INFO=mkl, BUILD_TYPE=Release, CUDA_VERSION=11.1, CUDNN_VERSION=8.0.5, CXX_COMPILER=/opt/rh/devtoolset-7/root/usr/bin/c++, CXX_FLAGS= -Wno-deprecated -fvisibility-inlines-hidden -DUSE_PTHREADPOOL -fopenmp -DNDEBUG -DUSE_KINETO -DUSE_FBGEMM -DUSE_QNNPACK -DUSE_PYTORCH_QNNPACK -DUSE_XNNPACK -DSYMBOLICATE_MOBILE_DEBUG_HANDLE -O2 -fPIC -Wno-narrowing -Wall -Wextra -Werror=return-type -Wno-missing-field-initializers -Wno-type-limits -Wno-array-bounds -Wno-unknown-pragmas -Wno-sign-compare -Wno-unused-parameter -Wno-unused-variable -Wno-unused-function -Wno-unused-result -Wno-unused-local-typedefs -Wno-strict-overflow -Wno-strict-aliasing -Wno-error=deprecated-declarations -Wno-stringop-overflow -Wno-psabi -Wno-error=pedantic -Wno-error=redundant-decls -Wno-error=old-style-cast -fdiagnostics-color=always -faligned-new -Wno-unused-but-set-variable -Wno-maybe-uninitialized -fno-math-errno -fno-trapping-math -Werror=format -Wno-stringop-overflow, LAPACK_INFO=mkl, PERF_WITH_AVX=1, PERF_WITH_AVX2=1, PERF_WITH_AVX512=1, TORCH_VERSION=1.9.0, USE_CUDA=ON, USE_CUDNN=ON, USE_EXCEPTION_PTR=1, USE_GFLAGS=OFF, USE_GLOG=OFF, USE_MKL=ON, USE_MKLDNN=ON, USE_MPI=OFF, USE_NCCL=ON, USE_NNPACK=ON, USE_OPENMP=ON, 

10/08 16:42:33 - mmengine - INFO - TorchVision: 0.10.0+cu111
10/08 16:42:33 - mmengine - INFO - OpenCV: 4.8.0
10/08 16:42:33 - mmengine - INFO - MMEngine: 0.7.4
10/08 16:42:33 - mmengine - INFO - MMCV: 2.0.1
10/08 16:42:33 - mmengine - INFO - MMCV Compiler: GCC 7.3
10/08 16:42:33 - mmengine - INFO - MMCV CUDA Compiler: 11.1
10/08 16:42:33 - mmengine - INFO - MMDeploy: 1.3.0+
10/08 16:42:33 - mmengine - INFO - 

10/08 16:42:33 - mmengine - INFO - **********Backend information**********
10/08 16:42:33 - mmengine - INFO - tensorrt:    None
10/08 16:42:33 - mmengine - INFO - ONNXRuntime: 1.15.1
10/08 16:42:33 - mmengine - INFO - ONNXRuntime-gpu:     None
10/08 16:42:33 - mmengine - INFO - ONNXRuntime custom ops:      Available
10/08 16:42:33 - mmengine - INFO - pplnn:       None
10/08 16:42:33 - mmengine - INFO - ncnn:        None
10/08 16:42:33 - mmengine - INFO - snpe:        None
10/08 16:42:33 - mmengine - INFO - openvino:    None
10/08 16:42:33 - mmengine - INFO - torchscript: 1.9.0+cu111
10/08 16:42:33 - mmengine - INFO - torchscript custom ops:      NotAvailable
10/08 16:42:33 - mmengine - INFO - rknn-toolkit:        None
10/08 16:42:33 - mmengine - INFO - rknn-toolkit2:       None
10/08 16:42:33 - mmengine - INFO - ascend:      None
10/08 16:42:33 - mmengine - INFO - coreml:      None
10/08 16:42:33 - mmengine - INFO - tvm: None
10/08 16:42:33 - mmengine - INFO - vacc:        None
10/08 16:42:33 - mmengine - INFO - 

10/08 16:42:33 - mmengine - INFO - **********Codebase information**********
10/08 16:42:33 - mmengine - INFO - mmdet:       3.0.0
10/08 16:42:33 - mmengine - INFO - mmseg:       1.1.0
10/08 16:42:33 - mmengine - INFO - mmpretrain:  None
10/08 16:42:33 - mmengine - INFO - mmocr:       None
10/08 16:42:33 - mmengine - INFO - mmagic:      None
10/08 16:42:33 - mmengine - INFO - mmdet3d:     None
10/08 16:42:33 - mmengine - INFO - mmpose:      None
10/08 16:42:33 - mmengine - INFO - mmrotate:    None
10/08 16:42:33 - mmengine - INFO - mmaction:    None
10/08 16:42:33 - mmengine - INFO - mmrazor:     None
10/08 16:42:33 - mmengine - INFO - mmyolo:      None

**Comment by RunningLeon** [comment]
that's a bit strange. Does it have something todo with onnxruntime version? Maybe you can provide your test image to reproduce.

**Comment by AnRanbel** [comment]
> that's a bit strange. Does it have something todo with onnxruntime version? Maybe you can provide your test image to reproduce.

1181x1181
![Landslide001](https://github.com/open-mmlab/mmdeploy/assets/33872705/6ecf8198-94eb-4df3-a09f-3c9a6df5bab6)
128x128
![Landslide001_128x128](https://github.com/open-mmlab/mmdeploy/assets/33872705/ccc64979-8873-4931-9d4d-424b303fc0f1)

**Comment by RunningLeon** [comment]
hi, cannot reproduce with the official ckpt. maybe you should check what you changed compared with the official model.
